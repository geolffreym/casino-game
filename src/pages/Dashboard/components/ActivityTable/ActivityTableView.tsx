import React, { FC } from 'react'
import Table, {
  TableBody, TableColumn, TableHeader,
  TableScroller, TableRow, TableRowEmpty
} from '@components/Table'
import { ActivityTableRow } from '@pages/Dashboard/components/ActivityTable/ActivityTableRow'

/* eslint-disable  @typescript-eslint/explicit-function-return-type */
export interface BalanceTableViewProps {
  rows: any[]
  columns: any[]
  slice: number[]
}

const ActivityTableView: FC<BalanceTableViewProps> = ({ columns, rows, slice }): JSX.Element => {
  return (
    <TableScroller>
      <Table stickyHeader aria-label='Balance table'>
        <TableHeader>
          <TableRow>
            {
              columns.map((column, i) =>
                <TableColumn key={i}>
                  {column.name}
                </TableColumn>
              )
            }
          </TableRow>
        </TableHeader>
        <TableBody>
          {
            (rows.slice(...slice).length > 0)
              ? rows.slice(...slice).map((row, i) => {
                return <ActivityTableRow {...row} key={i} />
              })
              : <TableRowEmpty />
          }
        </TableBody>
      </Table>
    </TableScroller>
  )
}

export default ActivityTableView
