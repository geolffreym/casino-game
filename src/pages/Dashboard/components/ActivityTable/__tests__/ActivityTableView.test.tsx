import { mount } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import ActivityTableView from '@pages/Dashboard/components/ActivityTable/ActivityTableView'

/* eslint-disable no-undef */
/* eslint-disable @typescript-eslint/consistent-type-assertions */
describe('<ActivityTableView/> ', () => {
  it('should render valid columns', () => {
    const rows = [
      {
        id: '1213',
        result: [0, 0, 0],
        time: '07/05/21'
      }
    ]
    const component = mount(<ActivityTableView rows={rows} columns={[]} slice={[0, 2]} />)

    expect(component).toMatchSnapshot()
  })
})
