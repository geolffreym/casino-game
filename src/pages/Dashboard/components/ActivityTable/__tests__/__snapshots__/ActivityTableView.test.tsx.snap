// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActivityTableView/>  should render valid columns 1`] = `ReactWrapper {}`;
